packages/js/components/src/experimental-select-control/select-control.tsx

```diff
@@ -103,20 +114,51 @@ export const SelectControl = ( {
 
 					break;
 				case useCombobox.stateChangeTypes.InputKeyDownEnter:
+				case useCombobox.stateChangeTypes.FunctionSelectItem:
 				case useCombobox.stateChangeTypes.ItemClick:
 				case useCombobox.stateChangeTypes.InputBlur:
 					if ( selectedItem ) {
 						onSelect( selectedItem );
-						setInputValue(
-							multiple ? '' : itemToString( selectedItem )
-						);
+						if ( clearSearchOnSelect ) {
```

Comment:
I have mixed feelings of this, but unless `clearSearchOnSelect` is a very common pattern we anticipate on using, I'd prefer to let the consumer dictate that.

We could provide an `onStateChange` to the consumer and `setInputValue` to allow better control over what happens with this select control.

Reply:
I could remove this one again, I originally thought I needed this for the category dropdown, but given the window will stay open, clearing the search would seem very odd.
I could only see some benefit for clearing on `InputBlur` but not for any of the other options.
But given that is not required for us, I can remove this for now.

Reply:
Actually, this logic is handy for us on a multi select, when using it with keep open. As things are a little strange when a user search for a value and selects one, then it auto clears showing a different list (while still open).

I could still remove the `clearSearchOnSelect` option, but make it not clear for `multiple` selection. Maybe only not clear for multiple selection and `keepMenuOpenOnSelect`